crates/biome_html_formatter/src/html/auxiliary/astro_frontmatter_element.rs

```diff
@@ -9,6 +10,17 @@ impl FormatNodeRule<HtmlAstroFrontmatterElement> for FormatHtmlAstroFrontmatterE
         node: &HtmlAstroFrontmatterElement,
         f: &mut HtmlFormatter,
     ) -> FormatResult<()> {
-        format_verbatim_node(node.syntax()).fmt(f)
+        let HtmlAstroFrontmatterElementFields {
+            l_fence_token,
+            content_token,
+            r_fence_token,
+        } = node.as_fields();
+
+        write!(f, [l_fence_token.format(), hard_line_break(),])?;
+
+        if let Some(content_token) = content_token {
+            write!(f, [content_token.format(), hard_line_break()])?;
+        }
+        write!(f, [r_fence_token.format(), hard_line_break()])
```

Comment:
... it appears you're already adding a hard line at the end, so idk what the issue is